python/tvm/meta_schedule/testing/validate_database.py

```diff
@@ -124,158 +252,495 @@ def default_input_generator(mod: IRModule) -> List[tvm.nd.NDArray]:
+        print(
+            f"Total {len(workloads)} unique original TIR to validate. "
+            f"Deduplicated in {float(profiler.get()['deduplicate records']): 3.3f} sec."
+        )
+        if ARGS.top_k < 10**9:
+            print(f"Top {ARGS.top_k} records for each original TIR will be validated.")
+            total = len(workloads) * ARGS.top_k
+        print()
+
+        # validate correctness
+        counter = 0
+        for item in workloads:
+            original_mod = item.mod
+            records = database.get_top_k(
+                workload=database.commit_workload(original_mod), top_k=ARGS.top_k
+            )
+            if len(records) < ARGS.top_k:
+                total -= ARGS.top_k - len(records)
+            inputs = to_numpy(check_and_run(ARGS.input_generator_func, original_mod))
+            original_res, original_run_secs = local_build_and_run(
+                original_mod,
+                target=ARGS.baseline_target,
+                inputs=inputs,
+                device=get_runtime_device(ARGS.baseline_target),
+            )
+            scheduled_mods = [_apply_trace(original_mod, record.trace) for record in records]
+            builder_results = _build_all_mods(scheduled_mods, builder, target)  # type: ignore
+            for i, record in enumerate(records):
+                counter += 1
+                print_result = print_with_counter_func(counter=counter, total=total)
+                if is_failed_record(record):
+                    # skip failed records where run_secs is 1e9
+                    # these records are only negative samples for cost model
+                    print_result(result="skip")
+                    continue
                 try:
-                    is_success = validate_correctness(
-                        original_mod=original_mod,
-                        scheduled_mod=scheduled_mod,
-                        target=target,
-                        baseline_target=ARGS.baseline_target,
-                        dev_type=dev_type,
+                    # prepare scheduled module
+                    scheduled_mod = scheduled_mods[i]
+                    # check build result
+                    builder_result = builder_results[i]
+                    _check_builder_result(builder_result)
+                    # fetch functions
+                    (
+                        f_with_args_alloc_argument,
+                        f_with_args_run_evaluator,
+                    ) = make_alloc_arg_and_check(
+                        inputs,
+                        original_mod,
+                        scheduled_mod,
+                        str(record.trace),
+                        original_res=original_res,
+                        original_run_secs=original_run_secs,
+                        print_result=print_result,
+                    )
+                    # create rpc runner
+                    runner = ms.runner.RPCRunner(
```

Comment:
Will it be useful to make rpc optional, which means it uses LocalRunner if no rpc related args are provided?

Reply:
I think that's a reasonable ask, what do you think would be a good user interface?
Say, `--rpc-host="local"`?

Reply:
Maybe just make the `--rpc-host` arg and its friends optional and use local runner if `--rpc-host` is not given by user

Reply:
Sure, can do that.

Reply:
I'll throw a warning though, to make sure user is aware of that.